src/query.rs

```diff
@@ -141,9 +141,80 @@ Failed to parse a query: {e}
     }
 }
 
+/// Configured values for a [`SemverQuery`] that differ from the lint's defaults.
+#[derive(Debug, Clone, Default, PartialEq, Eq)]
+pub struct QueryOverride {
+    /// The required version bump for this lint; see [`SemverQuery`].`required_update`.
+    ///
+    /// If this is `None`, use the query's default `required_update` when calculating
+    /// the effective required version bump.
+    pub required_update: Option<RequiredSemverUpdate>,
+
+    /// The lint level for this lint; see [`SemverQuery`].`lint_level`.
+    ///
+    /// If this is `None`, use the query's default `lint_level` when calculating
+    /// the effective lint level.
+    pub lint_level: Option<LintLevel>,
+}
+
+/// A mapping of lint ids to configured values that override that lint's defaults.
+pub type OverrideMap = BTreeMap<String, QueryOverride>;
+
+/// Stores a stack of [`OverrideMap`] references such that items towards the top of
+/// the stack have *higher* precedence and override items lower in the stack if both
+/// are set.
```

Comment:
Let's define what "top of the stack" is relative to our backing `Vec`.

I'm also not sure what `if both are set` refers to. Presumably the overriding behavior is only when two maps specify the same key, right? That would be my assumption if we don't say anything, so saying `if both are set` is making me think the answer is something more surprising instead..?
```suggestion
/// Stores a stack of [`OverrideMap`] references such that items towards the top of
/// the stack (later in the backing `Vec`) have *higher* precedence and override 
/// items lower in the stack.
```

Reply:
Yes.  I wasn't sure to phrase this succinctly, but precedence happens when, for a given key in `{required_update, lint_level}` and a given lint `id`, if both maps have an entry for the lint with id `id` and `override.key` is not `None` for both of them, the override found at the top of the stack will take precedence.

```suggestion
/// Stores a stack of [`OverrideMap`] references such that items towards the top of
/// the stack (later in the backing `Vec`) have *higher* precedence and override items lower in the stack.
/// That is, when an override is set and not `None` for a given lint in multiple maps in the stack, the value
/// at the top of the stack will be used to calculate the effective lint level or required version update.  
```

Does this change communicate it more clearly?

Reply:
Perfect!